buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/incubation/IncubatingApiAggregateReportTask.kt

```diff
@@ -0,0 +1,112 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.gradlebuild.buildquality.incubation
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.workers.IsolationMode
+import org.gradle.workers.WorkerExecutor
+import java.io.File
+import javax.inject.Inject
+
+
+@CacheableTask
+open class IncubatingApiAggregateReportTask @Inject constructor(private val workerExecutor: WorkerExecutor) : DefaultTask() {
+
+    @Input
+    var reports: Map<String, File>? = null
+
+    @OutputFile
+    val htmlReportFile = project.objects.fileProperty().also {
+        it.set(project.layout.buildDirectory.file("reports/incubation/all-incubating.html"))
+    }
+
+    @TaskAction
+    fun generateReport() {
+        workerExecutor.submit(GenerateReport::class.java) {
+            isolationMode = IsolationMode.CLASSLOADER
+            params(reports, htmlReportFile.asFile.get())
+        }
+    }
+}
+
+
+typealias ReportNameToProblems = MutableMap<String, MutableSet<String>>
+
+
+open class GenerateReport @Inject constructor(private val reports: Map<String, File>, private val outputFile: File) : Runnable {
+    override
+    fun run() {
+        val byVersion = mutableMapOf<String, ReportNameToProblems>()
+        reports.forEach { name, file ->
+            file.forEachLine(Charsets.UTF_8) {
+                val (version, _, problem) = it.split(';')
+                byVersion.getOrPut(version) {
+                    mutableMapOf()
+                }.getOrPut(name) {
+                    mutableSetOf()
+                }.add(problem)
+            }
+        }
+        generateReport(byVersion)
+    }
+
+    private
+    fun generateReport(data: Map<String, ReportNameToProblems>) {
+        outputFile.parentFile.mkdirs()
+        outputFile.printWriter(Charsets.UTF_8).use { writer ->
+            writer.println("""<html lang="en">
+    <head>
+       <META http-equiv="Content-Type" content="text/html; charset=UTF-8">
+       <title>Incubating APIs</title>
+       <link xmlns:xslthl="http://xslthl.sf.net" rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,400i,700">
+       <meta xmlns:xslthl="http://xslthl.sf.net" content="width=device-width, initial-scale=1" name="viewport">
+       <link xmlns:xslthl="http://xslthl.sf.net" type="text/css" rel="stylesheet" href="https://docs.gradle.org/current/userguide/base.css">
```

Comment:
Perhaps we could use the same Bootstrap-based CSS we use with the binary compatibility report... I can take a look at this in a separate PR.